frontend/src/components/workspaces/WorkspaceContainer/WorkspaceContainer.tsx

```diff
@@ -0,0 +1,131 @@
+import { FC, useState, useEffect } from 'react';
+import { UserSwitchOutlined, PlusOutlined } from '@ant-design/icons';
+import Button from 'antd-button-color';
+import { TemplatesTable } from '../Templates/TemplatesTable';
+import { TemplatesEmpty } from '../Templates/TemplatesEmpty';
+import Box from '../common/Box';
+
+export interface IWorkspaceContainerProps {
+  workspace: {
+    id: number;
+    title: string;
+    role: string;
+    templates: Array<{
+      id: string;
+      name: string;
+      gui: boolean;
+      instances: Array<{
+        id: number;
+        name: string;
+        ip: string;
+        status: boolean;
+      }>;
+    }>;
+  };
+}
+
+const WorkspaceContainer: FC<IWorkspaceContainerProps> = ({ ...props }) => {
+  const { workspace } = props;
+  const [templates, setTemplates] = useState(workspace.templates);
+
+  useEffect(() => {
+    setTemplates(workspace.templates);
+  }, [workspace.templates]);
+
+  const createInstance = (id: string) => {
+    setTemplates(templates => {
+      return templates.map(template =>
+        template.id === id
+          ? Object.assign({}, template, {
+              instances: [
+                ...template.instances,
+                {
+                  id: template.instances.length
```

Comment:
Why are you using hard-coded values

Reply:
Changed applied [here](https://github.com/netgroup-polito/CrownLabs/pull/460/commits/77c8d7a732e466080c9e305ba922259e2ecdaa42)